### PR TITLE
Update formula testing

### PR DESCRIPTION
**Pull request by epcim**
Pls. comment/suggest what to add/modify, etc.

### CODE REVIEW

File: doc/source/develop/testing-formulas.rst [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+  # install test kitchen
+  gem install test-kitchen
+
+An optional ``Gemfile`` in the main directory may contain fine tuned dependencies for specific workflows.
+To install Gefmfile dependencies run ``gem install bundler`` and then run ``bundle install [--path `pwd`/.vendor/bundle]``.
+
+
+
+
```
Comment (fpytloun) [inline doc/source/develop/testing-formulas.rst]: Why so many empty lines?

File: doc/source/develop/testing-formulas.rst [hunk 2]
```diff
@@ -0,0 +1,264 @@
+`Home <index.html>`_ SaltStack-Formulas Development Documentation
```
Comment (fpytloun) [inline doc/source/develop/testing-formulas.rst:1]: Link to home is not needed

File: doc/source/develop/testing-formulas.rst [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+
+Salt Formula testing
+====================
+
+Each formula contains *Makefile* with *test* target. Under ``tests`` directory are located resources for test execution.
```
Comment (fpytloun) [inline doc/source/develop/testing-formulas.rst]: This paragraph and few bellow are already part of `Formula smoke tests`

File: doc/source/develop/testing-formulas.rst [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+The purpose of the smoke test is to find syntax, typo issues and verify example pillar data against the formula.
+
+
+Jinja templates testing
+-----------------------
+To check jinja templates you may use:
+
+.. code-block:: shell
+
+  cat > check_my_jinja_recursive.py <<-EOF
```
Comment (fpytloun) [inline doc/source/develop/testing-formulas.rst]: IMO this is under indented

File: doc/source/develop/testing-formulas.rst [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+
+Additional resources.
+
+* https://github.com/chef/inspec
+* https://github.com/chef/kitchen-inspec
+
+
+Example verification scripts under `tests/integration` folder of the formula:
+
+.. code-block:: shell
```
Comment (fpytloun) [inline doc/source/develop/testing-formulas.rst]: Not a shell, use just `::`

File: doc/source/develop/testing-formulas.rst [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+  │           └── server_spec.rb
+  ├── pillar
+  │   ├── backupmx.sls
+  │   ├── relay.sls
+  │   └── server.sls
+  └── run_tests.sh
+
+
+Requirements
+^^^^^^^^^^^^^^^^^^
```
Comment (fpytloun) [inline doc/source/develop/testing-formulas.rst]: Too long underline. Run make html to see and fix warnings/error and final document.

File: doc/source/develop/testing-formulas.rst [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+
+Use latest stable kitchen-salt and kitchen-test.
+Minimal supported version of kitchen-salt is >= v0.0.25.
+
+**Intstall Procedure**
```
Comment (fpytloun) [inline doc/source/develop/testing-formulas.rst]: Typo

File: doc/source/develop/testing-formulas.rst [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+
+See http://kitchen.ci/ for more details.
+
+To install user side use:
```
Comment (fpytloun) [inline doc/source/develop/testing-formulas.rst]: Maybe instruct user to use bundler instead

**Comment by fpytloun** [comment]
Don't commit submodules.

File: doc/source/develop/testing-formulas.rst (hunk 2 above)
Comment (epcim) [reply doc/source/develop/testing-formulas.rst:1]: It's in the other files too..

**Pull request merged**